### ISSUE TITLE
Can't transfer folder + sub folders/files to new user?

### ISSUE THREAD

**Issue by scfigg** [taers232c/GAMADV-XTD#41]
Attempting to take specific folders from one user, and transfer them to a different user.

Here is the command I've been testing:

gam.exe user sourceuser@domain.com transfer drive targetuser@domain.com foldername

It shows this:

User: sourceuser@domain.com, Transfer 1 Drive Folder
Source drive size: 240mb, Target drive free: UNLIMITED
  User: sourceuser@domain.com, Drive File/Folder: foldername, Transfer Failed: Not Found

**Comment by scfigg** [comment]
or is there a better way of copying/moving specific folders to specific users? (not all data, just the specific folder and sub folders/files).

**Comment by taers232c** [comment]
Here is the Wiki: https://github.com/taers232c/GAMADV-XTD3/wiki/Users-Drive-Transfer#gam-direct-transfers <https://github.com/taers232c/GAMADV-XTD3/wiki/Users-Drive-Transfer#gam-direct-transfers>

That's not valid systax:
$ gam user testuser3 transfer drive testuser4 foldername
Command: /Users/Ross/Documents/GoogleApps/GAM3/gamadvxtd3.py user testuser3 transfer drive testuser4 >>>foldername<<< 

ERROR: Invalid argument
Help: Syntax in file /Users/Ross/Documents/GoogleApps/GAM3/GamCommands.txt
Help: Documentation is at https://github.com/taers232c/GAMADV-XTD3/wiki

What version of Gam do you have? Is that a copy and paste of the actual command?

Ross
--
ross.scroggs@gmail.com

**Comment by scfigg** [comment]
I have the latest windows release of GAMADVX. I can transfer all files from the user if I don't specify the folder. But I want to specify a folder name. I can make sense of the wiki for doing a specific folder/sub files. Can you provide an example of what that would be?

**Comment by taers232c** [comment]
gam.exe user sourceuser@domain.com <mailto:sourceuser@domain.com> transfer drive targetuser@domain.com <mailto:targetuser@domain.com> select name foldername

Do you understand all of the role retention options?

Ross
--
ross.scroggs@gmail.com

**Comment by scfigg** [comment]
Thanks - I was missing the "name" part after 'select'. It's working for me now. I can even do it in bulk with a CSV file, which is what I needed. 

Is there any way to force it to create a targetfoldername if it doesn't exist already? Right now it transfers it with "sourceuser@domain.com's old files" - which is fine - just curious if I can force it to rename the folder or put it into a new folder name.

**Comment by taers232c** [comment]
Yes, see: https://github.com/taers232c/GAMADV-XTD/wiki/Users-Drive-Transfer#gam-direct-transfers <https://github.com/taers232c/GAMADV-XTD/wiki/Users-Drive-Transfer#gam-direct-transfers>

Choose one of the following options to specify an existing target user folder as the parent folder of the subfolder. If neither option is chosen, the root folder of the target user will be the parent folder of the subfolder.

targetfolderid <DriveFolderID> - The ID of an existing folder owned by the target user,
targetfoldername <DriveFolderName> - The name of an existing folder owned by the target user.
Use the following option to specify the subfolder of the parent folder to receive the transferred files. This folder will be created if necessary. If targetuserfoldername is not specified, the default value #user# old files will be used.

targetuserfoldername <DriveFolderName> - The name of a subfolder under the parent folder.
targetuserfoldername "" - No subfolder will be created, the transferred files will be transferred to the parent folder.

Ross
--
ross.scroggs@gmail.com

**Comment by scfigg** [comment]
Awesome, thank you! I know I read that a dozen times last night - probably just too tired and didn't read it correctly. This works perfectly for what I need it for!
